src/chunk.c

```diff
@@ -880,31 +957,24 @@ chunk_fill_from_stub(Chunk *result, ChunkStub *chunk_stub, bool tuplock)
 				Anum_chunk_idx_id,
 				BTEqualStrategyNumber,
 				F_INT4EQ,
-				Int32GetDatum(chunk_stub->id));
+				Int32GetDatum(stubctx->stub->id));
 
-	num_found = ts_scanner_scan(&ctx);
+	num_found = ts_scanner_scan(&scanctx);
 
-	if (num_found != 1)
-		elog(ERROR, "no chunk found with ID %d", chunk_stub->id);
+	Assert(num_found == 0 || num_found == 1);
 
-	if (NULL == result->cube)
-		result->cube = ts_hypercube_from_constraints(result->constraints, CurrentMemoryContext);
-	else
+	if (stubctx->is_dropped)
+	{
+		Assert(num_found == 0);
+		return NULL;
+	}
 
-		/*
-		 * The hypercube slices were filled in during the scan. Now we need to
-		 * sort them in dimension order.
-		 */
-		ts_hypercube_slice_sort(result->cube);
+	if (num_found != 1)
```

Comment:
Not sure what is wrong with the current condition given that the only possible value here should be 0 or 1. So, if it is not 1, the chunk is not found.

Reply:
It's confusing due to the assert above. According to other places with similar condition, `num_found == 0` gives one error message, while `num_found > 1` gives another error message.

Reply:
Sorry, but I am not following how this is confusing. It is asserted that `num_found` can be either 1 or 0, so I don't understand why anything greater than 1 is important here?

In any case, I can change, but I think what you propose because the only viable value one returning from this function below is num_count==1, which is exactly what this error condition is saying (even without asserts).

Reply:
The error message is only suitable for `num_found == 0`. Thus I find it confusing. In any case I don't find it is so important, since the result of execution is correct.